### PR TITLE
PYIC-5284: Update identity check request for Nino

### PR DESCRIPTION
**Pull request by MikeCollingwood**
### What changed

- Updated state machine to take evidenceRequest and to convey that information to the BuildCriOauth output JAR
- Other code cleanup

### Why did it change

- We want to be able to specify the evidence request's scoring policy and strength required in for a more detailed request of the CRIs
- Sonarcloud now requires some code cleanup and other issues appear as warnings in committing

### Issue tracking

- [PYIC-5284](https://govukverify.atlassian.net/browse/PYIC-5284)

### Testing

- Having deployed the changes, manual tests were done where I temporarily added the updated state response to the passport cri state:
Screenshot 2024-03-14 at 11 37 09
Screenshot 2024-03-14 at 11 37 03

Here, you can see we see the evidence requested in the CRI stub.

[PYIC-5284]: https://govukverify.atlassian.net/browse/PYIC-5284?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java [hunk 1]
```diff
@@ -373,7 +382,7 @@ private EvidenceRequest getEvidenceRequestForF2F(List<VerifiableCredential> vcs)
             return null;
         }
 
-        return new EvidenceRequest(minViableStrengthOpt.getAsInt());
+        return new EvidenceRequest(null, minViableStrengthOpt.getAsInt());
```
Comment (blakeyp) [inline lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java:385]: Should be covered by regression tests but probably worth verifying these changes don't change the behaviour/format of what we're sending to F2F in the evidence claim

**Review by blakeyp — COMMENTED** [review]
🚀 looks good, suggest we hold this until Orange have implemented the respective changes in the NINO CRI (otherwise we'll break the integration we just got working 😅)

File: lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java (hunk 1 above)
Comment (blakeyp) [reply lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java:385]: Can see there are a few scenarios for this- https://github.com/govuk-one-login/ipv-core-tests/blob/d995435539503fdc926456034951856a578f5fcc/src/test/resources/features/Face2Face.feature#L251

File: .secrets.baseline [hunk 2]
```diff
@@ -75,10 +75,6 @@
     {
       "path": "detect_secrets.filters.allowlist.is_line_allowlisted"
     },
-    {
-      "path": "detect_secrets.filters.common.is_baseline_file",
-      "filename": ".secrets.baseline"
-    },
```
Comment (DanCorderIPV) [inline .secrets.baseline:81]: I've noticed that your PRs often remove this section from this file, and then other's put it back. Might be worth figuring out why and if it matters.

**Comment by Joe-Edwards-GDS** [comment]
Has this been superseded by #1918 - can we close?

**Pull request closed without merging**